tests/lib/location/src/location_test.c

```diff
@@ -246,6 +256,39 @@ void test_location_method_str(void)
 	TEST_ASSERT_EQUAL_STRING("Unknown", method);
 }
 
+/* Since CMock cannot return values by reference to variadic functions, we write our own mock
+ * function for this.
+ * For this test suite, this mock will always test the arguments against fixed values. It will
+ * always return a fixed System Mode.
+ */
+int __wrap_nrf_modem_at_scanf(const char *cmd, const char *fmt, ...)
```

Comment:
Related to this comment, I'm trying to see if there would be a more general solution:
https://github.com/nrfconnect/sdk-nrf/pull/8285#discussion_r969521833

Reply:
There is a generic Cmock for nrf_modem_at in PR #8766.
I think we should make a decision which way to go for mocking nrf_modem_at before we merge anything as this PR is not going to the same direction as PR #8766.